astropy/table/table.py

```diff
@@ -361,7 +363,12 @@ def __init__(self, data=None, masked=None, names=None, dtype=None,
             n_cols = len(default_names)
 
         elif isinstance(data, Table):
-            init_func = self._init_from_table
+            # If input meta is not None (default), then provide direction to
+            # self._init_from_table to entirely ignore meta on the input table.
+            # This saves a potential deepcopy of the table meta that later gets
+            # discarded.
+            use_table_meta = (meta is None)
+            init_func = functools.partial(self._init_from_table, use_table_meta=use_table_meta)
```

Comment:
I wonder if we aren't better off just moving the few lines from `_init_from_table` here and use `init_func = self._init_from_list`...

Reply:
Done.